src/chart/helper/multipleGraphEdgeHelper.js

```diff
@@ -0,0 +1,221 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+var linkMap = {};
+var curvenessList = [];
+
+/**
+ * params handler
+ * @param seriesModel
+ * @returns {*}
+ */
+var getAutoCurvenessParams = function (seriesModel) {
+    var autoCurveness = seriesModel.getModel('autoCurveness');
+
+    if (!autoCurveness || !autoCurveness.option) {
+        return null;
+    }
+
+    return autoCurveness.option;
+};
+
+/**
+ * Generate a list of edge curvatures, 20 is the default
+ * @param seriesModel
+ * @param addLength append length
+ * @return  20 => [0, -0.2, 0.2, -0.4, 0.4, -0.6, 0.6, -0.8, 0.8, -1, 1, -1.2, 1.2, -1.4, 1.4, -1.6, 1.6, -1.8, 1.8, -2]
+ */
+var createCurveness = function (seriesModel, addLength) {
+    var autoCurvenssParmas = getAutoCurvenessParams(seriesModel);
+    var length = 20;
+
+    // handler the function set
+    if (typeof autoCurvenssParmas === 'number') {
+        length = autoCurvenssParmas;
+    }
+    else if (typeof autoCurvenssParmas === 'function') {
+        curvenessList = autoCurvenssParmas();
+        return;
+    }
+
+    // addLength
+    if (addLength) {
+        length = addLength;
+    }
```

Comment:
I'm not sure, but I think this "append length" logic is not needed.
It's more clear and safe to recreate `curvenessList` totally if any data changed.

Reply:
right，i think if the curvenessList is create by Internal code, and the ```autoCurveness``` is not set as function, the logic is not need